smallrye-reactive-messaging-in-memory/src/main/java/io/smallrye/reactive/messaging/connectors/InMemoryConnector.java

```diff
@@ -48,40 +45,79 @@
      *
      *     //...
      *
+     *     &#64;Before
+     *     public void setup() {
+     *         InMemoryConnector.switchIncomingChannelsToInMemory("my-channel");
+     *     }
+     *
+     *     // ..
+     *
      *     InMemorySource&lt;Integer&gt; channel = connector.source("my-channel");
      *     channel.send(1);
      *     channel.send(2);
      *
      * </pre></code>
      * <p>
-     * With the {@link InMemorySource}, you can send messages to the channel, mocking the incoming messages.
      *
-     * You can also retrieve an {@link InMemorySink} using:
+     * @param channels the channels to switch, must not be {@code null}, must not contain {@code null}, must not contain
+     *        a blank value
+     * @return The map of properties that have been defined. The method sets the system properties, but give
+     *         you this map to pass the properties around if needed.
+     */
+    public static Map<String, String> switchIncomingChannelsToInMemory(String... channels) {
```

Comment:
Do the methods need to return a `Map` of the values being set?

Given they're set as system properties, what's the use case?

Reply:
Quarkus Test Resources when using native test.